packages/dds/tree/src/feature-libraries/flex-map-tree/mapTreeNode.ts

```diff
@@ -790,6 +794,20 @@ function unboxedUnion<TTypes extends FlexAllowedTypes>(
 		}
 		return getOrCreateChild(mapTree, type, parent) as FlexTreeUnboxNodeUnion<TTypes>;
 	}
+	if (schema.allowedTypeSet !== "Any") {
```

Comment:
Use the const `Any` rather than a string here (that way it more maintainable, you can navigate to the docs, etc).

If we keep this logic rather than recomputing this on every access, it should be computed once in the FlexTreeFIeldSchema then just reused.

I don't think this is quite the policy change we want to make. It helps, but it doesn't fully align with simple-tree's desires.

If we have a leaf and non leaf and non-leaf type in a union, ideally we would unbox the leaf.

Related:
The exact details of when this unboxes are described in the schema aware typing in `FlexTreeUnboxNodeUnion`. If you change unboxing policy at all, that must be updated to match.